Urlopik/src/main/Modifier.java

```diff
@@ -49,36 +50,36 @@ else if ((mode==2) && (planned>0)){
 				start = LocalDate.parse(sc.nextLine());
 				System.out.print("Podaj koniec w formacie RRRR-MM-DD: ");
 				end = LocalDate.parse(sc.nextLine());
-			
-			
+
+
 				validD = dateValidator(start, end);											//checking the validity of dates (not from the past, in proper order and so on)
-				
-				if (mode==1) {
-					validA = allowanceValidator(start.until(end, ChronoUnit.DAYS), remaining);			//checking if remaining allowance is larger than planned off days
+
+				if (mode == OperationMode.SET) {
+					int plannedWorking = calcEstimatedOffLength(yearMap, start, end);
+					validA = allowanceValidator(plannedWorking, remaining);			//checking if remaining allowance is larger than planned off days
 				}
 				else validA=true;
-			
+
 			} catch (DateTimeParseException e) {
 				System.out.println("Błędny format daty!");
 				trigger = true;
 			}
-		
+
 		} while (validD==false || validA==false || trigger==true);

```

Comment:
Z reguły jak jest dużo parametrów w jednym if'ie to robi sie  z tego metode typu shouldDoSth(parametr). Potem taką metodę mozna sobie przetestowac osobno